src/Eloquent/Model.php

```diff
@@ -315,19 +316,6 @@ protected function fromDecimal($value, $decimals)
         return new Decimal128($this->asDecimal($value, $decimals));
     }
 
-    /** @inheritdoc */
-    protected function castAttribute($key, $value)
-    {
-        $castType = $this->getCastType($key);
-
-        return match ($castType) {
-            'immutable_custom_datetime','immutable_datetime' => str_starts_with($this->getCasts()[$key], 'immutable_date:') ?
-                $this->asDate($value)->toImmutable() :
-                $this->asDateTime($value)->toImmutable(),
```

Comment:
The cast types `immutable_custom_datetime` and `immutable_datetime` are already supported by the parent method: [`HasAttribute::castAttribute()`](https://github.com/laravel/framework/blob/v10.46.0/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php#L780-L782)

Reply:
- `$this->getCasts()[$key]` is what is defined in the `$casts` property (`date:j.n.Y H:i`).
- `$this->getCastType($key)` is a classification (`custom_datetime`)